File: Day03/Medium/Solution.c
#include <stdio.h>

void insertionSort(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        int ele = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > ele) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = ele;
    }
}

int main() {
    int n;
    printf("Enter number of elements: ");
    scanf("%d", &n);

    int arr[n];
    printf("Enter elements: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    insertionSort(arr, n);

    printf("Sorted array: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: Day03/Medium/Solution.py
class Solution:
    def insertionSort(self, arr):
        n = len(arr)
        for i in range(1, n):
            ele = arr[i]
            j = i - 1
            while j >= 0 and arr[j] > ele:
                arr[j + 1] = arr[j]
                j -= 1
            arr[j + 1] = ele

if __name__ == "__main__":
    sol = Solution()
    n = int(input("Enter number of elements: "))
    arr = list(map(int, input("Enter elements: ").split()))

    sol.insertionSort(arr)

    print("Sorted array:", *arr)
